### PR TITLE
Have a configurable maximum active work per peer

### PR DESCRIPTION
**Pull request by aschmahmann**

### CODE REVIEW

File: peertracker/peertracker.go [hunk 1]
```diff
@@ -172,6 +175,16 @@ func (p *PeerTracker) PopTasks(targetMinWork int) ([]*peertask.Task, int) {
 	var out []*peertask.Task
 	work := 0
 	for p.taskQueue.Len() > 0 && p.freezeVal == 0 && work < targetMinWork {
+		if p.maxActiveWorkPerPeer > 0 {
+			// Do not add work to a peer that is already maxed out
+			p.activelk.Lock()
```
Comment (Stebalien) [inline peertracker/peertracker.go:180]: NOTE: I think we need to revisit this lock.

Comment (Stebalien) [reply peertracker/peertracker.go:180]: But not now.

**Review by Stebalien — APPROVED** [review]
This _looks_ correct.

**Comment by marten-seemann** [comment]
Please rebase this PR on top of master, so that the new GitHub Actions workflows (merged in #13) are run.

**Review by petar — APPROVED** [review]
LGTM

**Pull request merged**